app/routes.js

```diff
@@ -326,6 +327,21 @@ export const getRoutes = (appContext) => {
   return (
     <Route path='/' {...props} render={routeProps => (
       <AppComponent {...routeProps} {...props}>
+        <Route
+          path="/"
+          render={({ location }) => {
+            if (ReactGA.isInitialized) {
+              ReactGA.send({
+                hitType: 'pageview',
+                page: location?.pathname + location?.search + location?.hash,
+                'page_search': location.search,
+                'page_hash': location.hash,
+              });
+            }
+            return null;
+          }}
+        />
```

Comment:
While I do like the concept of tracking the page views at the route level, I'm not a huge fan of having a `/` level `Route` that's doing duty as a makeshift middleware and always returns `null`. I think that using the `useLocation` hook in the `AppComponent` might be a cleaner/more idiomatic way to go: https://v5.reactrouter.com/web/api/Hooks/uselocation

Reply:
I could also use the `withRouter` HOC, combined with `history.listen` directly in app.js.  Something like: https://stackoverflow.com/a/45373907

Reply:
I think I may have played around with this approach as well (I think I had access to the history within the location prop, IIRC) and had it working with the component lifecycle methods, but had opted for the vanity route method (had found a couple examples of doing that online, but I get the distaste for a 'fake' route, though it seemed like having the logic within the routing section was a nice location)